portfolio.py

```diff
@@ -2,21 +2,49 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, target=None, np_array=None):
         dict.__init__(self)
-        self.date = date
+        if target is not None:
+            if isinstance(np_array, np.ndarray):
+                self.decode_nparray(target, np_array)
+            else:
+                raise TypeError(
+                    'The input is %s, not a numpy array.' %(type(np_array)))
```

Comment:
1. If you have only one parameter after `%`, you can ignore the pair of brackets.
2. Please separate `%` and paramters by a space.